iotlabmqtt/iotlabapi.py

```diff
@@ -187,9 +193,18 @@ def node_command(self, command, cmd_opt, archi, *nums):
 
     def _node_command(self, command, cmd_opt, archi, *nums):
         import iotlabcli.node
+        import iotlabsshcli.open_a8
         nodes = self._nodes_for_num(archi, *nums)
-        result = iotlabcli.node.node_command(self.api, command, self.expid,
-                                             nodes, cmd_opt)
+
+        if archi == 'm3':
+            result = iotlabcli.node.node_command(self.api, command, self.expid,
+                                                 nodes, cmd_opt)
+        elif archi =='a8':
```

Comment:
here you could have another class in a python file that implements the commands
from https://github.com/iot-lab/cli-tools/blob/master/iotlabcli/node.py

If you use ssh to do ssh commands, you can have them in that for, or have run equivalent scripts in IoT-LAB:
```
A8M3_OPENOCD_TEMPLATE_CMD = """
/usr/local/bin/openocd 
        -f "/etc/iotlab-flash-scripts/iot-lab-a8-m3.cfg"
        -f "target/stm32f1x.cfg"
        -c "init"
        -c "targets"
        {commands}
        -c "shutdown"
""".replace("\n", "").replace("\r", "").strip()

A8M3_OPENOCD_RESET_RUN_CMD = A8M3OPENOCD_TEMPLATE_CMD.format(
    commands=' -c "reset run"')
A8M3_OPENOCD_RESET_HALT_CMD = A8M3_OPENOCD_TEMPLATE_CMD.format(
    commands=' -c "reset halt"')
```